wordpress-mobile/WordPress-iOS
# Plugins persistence.

This is last remaining issue for plugins, meaning this would effectively close #8042.

Most of the code is making `PluginStoreState` (and all the objects contained therein) conform to `Codable`, so that we can save it to the disk.

The actual interesting part happens in two places: `PluginStore+Persistence.swift` and in two places in PluginStore: on `init(dispatcher:_)` and in `reinitialiseIfNeeded()`.

It's all deliberately super simple in it's approach, to see if this looks like a sensible way forward. From testing it seemed a-ok, but we'll see how it behaves "live".

To test:
1. Go to plugin directory
2. Wait for all the requests to finish
3. Tap the "back" button
4. Kill the app
5. Restart the app
6. Go offline
7. Go to plugin directory
8. Verify that plugins are displaying correctly, even though you're offline.
